cmd/osm-controller/osm-controller_test.go

```diff
@@ -2,74 +2,246 @@ package main
-
-			actual, err := k8sClient.CoreV1().Secrets(namespace).Get(context.Background(), secretName, v1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			expected := "-----BEGIN CERTIFICATE-----\nMIID"
-			stringPEM := string(actual.Data[constants.KubernetesOpaqueSecretCAKey])[:len(expected)]
-			Expect(stringPEM).To(Equal(expected))
-
-			expectedRootCert, err := certManager.GetRootCertificate()
-			Expect(err).ToNot(HaveOccurred())
-			Expect(actual.Data[constants.KubernetesOpaqueSecretCAKey]).To(Equal(expectedRootCert.GetCertificateChain()))
-		})
-	})
-})
-
-var _ = Describe("Test joining of URL paths", func() {
-	It("should correctly join URL paths", func() {
-		final := joinURL("http://foo", "/bar")
-		Expect(final).To(Equal("http://foo/bar"))
-	})
-
-	It("should correctly join URL paths", func() {
-		final := joinURL("http://foo/", "/bar")
-		Expect(final).To(Equal("http://foo/bar"))
-	})
-
-	It("should correctly join URL paths", func() {
-		final := joinURL("http://foo/", "bar")
-		Expect(final).To(Equal("http://foo/bar"))
-	})
-
-	It("should correctly join URL paths", func() {
-		final := joinURL("http://foo", "bar")
-		Expect(final).To(Equal("http://foo/bar"))
-	})
-})
+type FakeDebugServer struct {
+	stopCount  int
+	startCount int
+	stopErr    error
+
+	wg *sync.WaitGroup
+}
+
+func (f *FakeDebugServer) Stop() error {
+	f.stopCount++
+	f.wg.Done()
+	if f.stopErr != nil {
+		return errors.Errorf("Debug server error")
+	}
+	return nil
+}
+
+func (f *FakeDebugServer) Start() {
+	f.startCount++
+	f.wg.Done()
+}
+
+func mockDebugConfig(mockCtrl *gomock.Controller) *debugger.MockDebugServer {
+	mockDebugConfig := debugger.NewMockDebugServer(mockCtrl)
+	mockDebugConfig.EXPECT().GetHandlers().Return(map[string]http.Handler{
+		validRoutePath: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}),
+	}).AnyTimes()
+	return mockDebugConfig
+}
+
+func setupComponents(namespace, configMapName string, initialDebugServerEnabled bool, stop chan struct{}) (*testclient.Clientset, v1.ConfigMap, configurator.Configurator, error) {
+	kubeClient := testclient.NewSimpleClientset()
+
+	defaultConfigMap := map[string]string{
+		"enable_debug_server": strconv.FormatBool(initialDebugServerEnabled),
+	}
+	configMap := v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      configMapName,
+		},
+		Data: defaultConfigMap,
+	}
+	_, err := kubeClient.CoreV1().ConfigMaps(namespace).Create(context.TODO(), &configMap, metav1.CreateOptions{})
+	if err != nil {
+		return kubeClient, configMap, nil, err
+	}
+	cfg := configurator.NewConfigurator(kubeClient, stop, namespace, configMapName)
+	return kubeClient, configMap, cfg, nil
+}
+
+func TestConfigureDebugServerStart(t *testing.T) {
+	assert := assert.New(t)
+
+	// set up a controller
+	mockCtrl := gomock.NewController(t)
+	stop := make(chan struct{})
+
+	kubeClient, _, cfg, err := setupComponents(testOSMNamespace, testOSMConfigMapName, false, stop)
+	assert.Nil(err)
```

Comment:
Because this is a set up step and not really a test assertion itself, we should use something like t.Fatal() instead of `assert.Nil(error)`.